src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -41,14 +40,16 @@ public TypeFactory(OutputLibrary library, Type unknownType)
         public CSharpType CreateType(InputType inputType) => inputType switch
         {
             InputLiteralType literalType => CSharpType.FromLiteral(CreateType(literalType.LiteralValueType), literalType.Value),
-            InputUnionType unionType => CSharpType.FromUnion(unionType.UnionItemTypes.Select(CreateType).ToArray(), unionType.IsNullable),
+            InputUnionType unionType => CSharpType.FromUnion(unionType.UnionItemTypes.Select(x => CreateType(x)).ToArray(), unionType.IsNullable),
             InputListType { IsEmbeddingsVector: true } listType => new CSharpType(typeof(ReadOnlyMemory<>), listType.IsNullable, CreateType(listType.ElementType)),
             InputListType listType => new CSharpType(typeof(IList<>), listType.IsNullable, CreateType(listType.ElementType)),
             InputDictionaryType dictionaryType => new CSharpType(typeof(IDictionary<,>), inputType.IsNullable, typeof(string), CreateType(dictionaryType.ValueType)),
+            // TODO: consolidate the behavior when we handle type nullability in DPG
```

Comment:
do we have an issue for this?

Reply:
I thought we already have an issue regarding nullability for property and type

Reply:
Lets link the issue here for reference.

Reply:
https://github.com/Azure/autorest.csharp/issues/4701

Reply:
put the work item in the comment

Reply:
drop the TODO and always call WithNullable